zebra-chain/src/work/arbitrary.rs

```diff
@@ -17,3 +19,81 @@ impl Arbitrary for equihash::Solution {
 
     type Strategy = BoxedStrategy<Self>;
 }
+
+impl Arbitrary for CompactDifficulty {
+    type Parameters = ();
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        (vec(any::<u8>(), 32))
+            .prop_filter_map("zero CompactDifficulty values are invalid", |v| {
```

Comment:
This worries me. Proptest pretty aggressively advises against using any of the `filter` variants, and I'm guessing this one is going to return `None` quite often, so this might break our proptests or make them super slow. I'm guessing there's no easy way to generate random difficulties though...

Reply:
Note that `ExpandedDifficulty::to_compact` is infallible, and always returns a `CompactDifficulty`.

Reply:
Ping @yaahc - I'd like to finish up this PR soon, what do you think of my explanations here?

Reply:
sounds good